src/Loop.php

```diff
@@ -33,6 +47,20 @@ public static function execute(callable $callback, LoopDriver $driver)
     }
 
     /**
+     * Create a new driver if a factory is present, otherwise throw.
+     *
+     * @throws \LogicException if no factory is set
+     */
+    private static function createDriver()
+    {
+        if (self::$factory === null) {
+            throw new \LogicException("Can't create an event loop driver without a factory.");
+        }
+
+        return self::$factory->create();
```

Comment:
The line I linked to is a method that gets called on a user-defined object. If the user-defined object returns an invalid value for `create()` then it can break other internal code that depends on `create()` returning an implementation of `LoopDriver`. That is how it applies to "people being able to define their own".